src/Interpreters/MutationsInterpreter.cpp

```diff
@@ -1386,6 +1387,10 @@ void MutationsInterpreter::validate()
         }
     }
 
+    if (context->getSettingsRef()[Setting::validate_mutation_query])
+        // Make sure the mutation query is valid
+        prepareQueryAffectedQueryTree(commands, source.getStorage(), context);
```

Comment:
My only concern is what if a mutation doesn't work with the new analyzer because of an analyzer pass and we thus user decided to disable analyzer. In `prepareQueryAffectedQueryTree` we force the usage of `QueryTreePassManager` regardless whether the new analyzer is enabled or disabled.

What do you think about it?

Reply:
Good point and the new analyzer is disabled for mutations: https://github.com/ClickHouse/ClickHouse/blob/a15e5b72afae60e58c03cc517109d1006958ec8a/src/Interpreters/MutationsInterpreter.cpp#L438-L441

I'll make it work without it as well.